cassandra-scheduler/src/test/java/com/mesosphere/dcos/cassandra/scheduler/offer/ClusterTaskOfferRequirementProviderTest.java

```diff
@@ -0,0 +1,317 @@
+                Resources.getResource("scheduler.yml").getFile());
+
+        Identity initial = config.getIdentity();
+
+        curatorConfig = CuratorFrameworkConfig.create(server.getConnectString(),
+                10000L,
+                10000L,
+                Optional.empty(),
+                250L);
+
+        clusterTaskConfig = config.getClusterTaskConfig();
+
+        persistence = (ZooKeeperPersistence) ZooKeeperPersistence.create(
+                initial,
+                curatorConfig);
+
+        curator = persistence.getCurator();
+
+        identity = new IdentityManager(
+                initial,
+                persistence,
+                Identity.JSON_SERIALIZER);
+
+        identity.register("test_id");
+
+        configuration = new ConfigurationManager(
+                config.getCassandraConfig(),
+                config.getClusterTaskConfig(),
+                config.getExecutorConfig(),
+                config.getServers(),
+                config.getSeeds(),
+                "NODE",
+                config.getSeedsUrl(),
+                config.getDcUrl(),
+                config.getExternalDcsList(),
+                config.getExternalDcSyncMs(),
+                persistence,
+                CassandraConfig.JSON_SERIALIZER,
+                ExecutorConfig.JSON_SERIALIZER,
+                ClusterTaskConfig.JSON_SERIALIZER,
+                IntegerStringSerializer.get()
+        );
+
+        path = "/cassandra/" + config.getIdentity().getName() +"/tasks";
+
+        provider = new ClusterTaskOfferRequirementProvider(identity, cassandraTasks);
+    }
+
+    @After
+    public void afterEach() {
+
+        try {
+            curator.delete().deletingChildrenIfNeeded().forPath(path);
+        } catch (Exception e) {
+
+        }
+    }
+
+    @AfterClass
+    public static void afterAll() throws Exception {
+        persistence.stop();
+        server.close();
+        server.stop();
+    }
+
+    @Test
+    public void testConstructor() throws Exception {
+        ClusterTaskOfferRequirementProvider provider = new ClusterTaskOfferRequirementProvider(
+            identity, cassandraTasks);
+        CassandraDaemonTask task = cassandraTasks.createDaemon("test-daemon");
+        Protos.TaskInfo taskInfo = task.getTaskInfo();
+
+        OfferRequirement requirement = provider.getNewOfferRequirement(taskInfo);
+        Assert.assertNotNull(requirement);
+    }
+
+    @Test
+    public void testGetNewOfferRequirement() throws Exception {
+        OfferRequirement requirement = provider.getNewOfferRequirement(testTaskInfo);
+        Protos.TaskInfo taskInfo = requirement.getTaskRequirements().iterator().next().getTaskInfo();
+        Assert.assertEquals(taskInfo.getName(), "test-daemon");
+        Assert.assertTrue(taskInfo.getTaskId().getValue().contains("test-daemon"));
+        Assert.assertEquals(taskInfo.getSlaveId().getValue(), "");
+
+				List<Protos.Resource> resources = taskInfo.getResourcesList();
```

Comment:
Something's not right with whitespace here. Maybe github has messed up the spacing.